src/libs/API.js

```diff
@@ -92,35 +112,12 @@ Network.registerParameterEnhancer(addDefaultValuesToParameters);
  * Function used to handle expired auth tokens. It re-authenticates with the API and
  * then replays the original request
  *
- * @param {String} originalCommand
- * @param {Object} [originalParameters]
- * @param {String} [originalType]
- * @returns {Promise}
+ * @param {Object} originalRequest
  */
-function handleExpiredAuthToken(originalCommand, originalParameters, originalType) {
-    // When the authentication process is running, and more API requests will be requeued and they will
-    // be performed after authentication is done.
-    if (isAuthenticating) {
-        return Network.post(originalCommand, originalParameters, originalType);
-    }
-
-    // Prevent any more requests from being processed while authentication happens
-    Network.pauseRequestQueue();
-    isAuthenticating = true;
-
+function handleExpiredAuthToken(originalRequest) {
     // eslint-disable-next-line no-use-before-define
-    return reauthenticate(originalCommand)
-        .then(() => {
-            // Now that the API is authenticated, make the original request again with the new authToken
-            const params = addDefaultValuesToParameters(originalCommand, originalParameters);
-            return Network.post(originalCommand, params, originalType);
-        })
-        .catch(() => (
-
-            // If the request did not succeed because of a networking issue or the server did not respond requeue the
-            // original request.
-            Network.post(originalCommand, originalParameters, originalType)
-        ));
+    reauthenticate(originalRequest.command)
+        .finally(() => retry(originalRequest));
```

Comment:
Pausing the queue and setting the `isAuthenticating` flag is moved to `Authenticate` - when `Authenticate` is called it would raise the flag, and set it back to `false` once it's done

Calling `addDefaultValuesToParameters` is not necessary `Network.post` already does it

`if (isAuthenticating)`, then and catch are no longer necessary - we call reauthenticate and finally retry the original request
I used `finally` since the original logic retries the request on failures as well. This is so because it probably triggers re-authenticate again...

Reply:
"queue pausing" is a concept that I'd recommend we get rid of entirely. It's confusing and basically "code" for whether we are authenticating or not.

Reply:
If there's a way to get rid of pausing, the change here would ease it, as everything related was moved here

The original code was pausing the queue - I didn't change that but cleaned it up and provided tests to confirm it still works as expected